src/ActivityLogger.php

```diff
@@ -2,129 +2,123 @@
 
 namespace Spatie\Activitylog;
 
-use Illuminate\Auth\AuthManager;
 use Illuminate\Contracts\Config\Repository;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Support\Arr;
 use Illuminate\Support\Carbon;
 use Illuminate\Support\Str;
 use Illuminate\Support\Traits\Macroable;
 use Spatie\Activitylog\Contracts\Activity as ActivityContract;
-use Spatie\Activitylog\Exceptions\CouldNotLogActivity;
 
 class ActivityLogger
 {
     use Macroable;
 
-    /** @var \Illuminate\Auth\AuthManager */
-    protected $auth;
+    protected string $defaultLogName = '';
 
-    protected $defaultLogName = '';
+    protected CauserResolver $causerResolver;
 
-    /** @var string */
-    protected $authDriver;
+    protected ActivityLogStatus $logStatus;
 
-    /** @var \Spatie\Activitylog\ActivityLogStatus */
-    protected $logStatus;
+    protected ?ActivityContract $activity = null;
 
-    /** @var \Spatie\Activitylog\Contracts\Activity */
-    protected $activity;
+    protected ActivityLoggerBatch $batch;
 
-    public function __construct(AuthManager $auth, Repository $config, ActivityLogStatus $logStatus)
+    public function __construct(Repository $config, ActivityLogStatus $logStatus, ActivityLoggerBatch $batch, CauserResolver $causerResolver)
     {
-        $this->auth = $auth;
+        $this->causerResolver = $causerResolver;
 
-        $this->authDriver = $config['activitylog']['default_auth_driver'] ?? $auth->getDefaultDriver();
+        $this->batch = $batch;
 
         $this->defaultLogName = $config['activitylog']['default_log_name'];
 
         $this->logStatus = $logStatus;
     }
 
-    public function setLogStatus(ActivityLogStatus $logStatus)
+    public function setLogStatus(ActivityLogStatus $logStatus): self
```

Comment:
Isn't `static` the more correct return-type here? The user could override/extend the class and in that case `$this` will be the users class and not this one.
With PHP8.0 we can use this feature.

Reply:
Do I have to change all `self` return type to `static` ?

Reply:
Haven't checked all - but for most it will be correct. All `self` which class is bound to container should use `static`.
In fact `static` nearly never hurts.